.github/workflows/ci.yml

```diff
@@ -99,13 +99,58 @@ jobs:
         name: windows-native-library
         path: bin/ParquetSharpNative.dll
         
-  # Download both native shared libraries,
-  # Rebuild .NET projects (will automatically pick up both libraries and create a nuget package).
+  # Build and test everything on macOS.
+  # Upload the native shared library as an artifact.
+  native-osx:
```

Comment:
native-osx and native-linux look very similar. I wonder if they can be matrixed (with fail-safe = false) to reduce complexity here. I think we agree that the Windows one is too different.

Reply:
Good point, I will try to implement it that way.

Reply:
I meant fail-fast BTW, not fail-safe. As blogged about here: https://www.edwardthomson.com/blog/github_actions_6_fail_fast_matrix_workflows.html